### PR TITLE
opencv: enable videoio on darwin

### LINKED ISSUE #97575: OpenCV videoio Darwin/Mac issue/discussion
**Describe the issue**
In the opencv build for mac, why is `videoio` off by default?

https://github.com/NixOS/nixpkgs/blob/193a6a2307b7b29aa11bee309d4aa41840686ab0/pkgs/development/libraries/opencv/4.x.nix#L258-L261

What needs to be done to enable it? (either development work, dependencies that aren't yet nixpkgs, or, if possible, how to manually override the argument and simply set to `=ON`)

**To Reproduce**
See above

**Expected behavior**
There may be a good reason it's disabled, so I can't say that I'd expect videoio to be enabled by default.

**Notify maintainers**
@basvandijk

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 16.7.0, macOS 10.12.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.7`
 - channels(root): `"nixpkgs-20.09pre234801.1facbd61b17"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change
Fixes https://github.com/NixOS/nixpkgs/issues/97575
cc: @jeff-hykin @mjlbach

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/opencv/4.x.nix [hunk 1]
```diff
@@ -255,7 +255,7 @@ stdenv.mkDerivation {
   ] ++ lib.optionals stdenv.isDarwin [
     "-DWITH_OPENCL=OFF"
     "-DWITH_LAPACK=OFF"
-    "-DBUILD_opencv_videoio=OFF"
+    "-DBUILD_opencv_videoio=ON"
```
[the remark this answers is not in the document]
Comment (marsam) [reply pkgs/development/libraries/opencv/4.x.nix]: fixed, thanks!

**Comment by mjlbach** [comment]
@jeff-hykin Can you test? I don't use this functionality.

**Comment by jeff-hykin** [comment]
Yeah, I'm happy too. I might need to wait till next week since I'm got deadlines atm.

The bigger barrier might be, I'm not sure how point nix towards a specific branch/commit. So advice on that would be appreciated, but I can probably figure it out with the docs if needed. I have to learn it anyways for a separate nix issue.

**Comment by jeff-hykin** [comment]
I'll need python37 to test with it. e.g. `nix-shell -I nixpkgs=./nixpkgs -p opencv4` isn't enough for me to test. Copying from my shell.nix, I tried `nix-shell -I nixpkgs=./nixpkgs -p python37.opencv4` but that failed so I'm going to have to spend more time on it another day. 

`nix-shell -I nixpkgs=./nixpkgs -p opencv4` did compile `opencv` for me on Sierra without errors though

**Comment by mjlbach** [comment]
@jeff-hykin There are a lot of ways to test this PR. You might want to look into [nixpkgs-review](https://github.com/Mic92/nixpkgs-review)

cd to your local checkout of nixpkgs, and run the following command to test:

```
nixpkgs-review pr --checkout commit  97633 -p opencv4 -p python3Packages.opencv4
```

You can also just run the following to get a nix-shell with the package available:

```
nix-shell -I nixpkgs=https://github.com/marsam/nixpkgs/archive/1966a0a4e1c1e07991406e3b90e3b088df252aea.tar.gz -p opencv4 python3Packages.opencv4
```

**Comment by marsam** [comment]
I'm merging this as-is, but feel free to open an issue if you find an issue

**Comment by jeff-hykin** [comment]
> You can also just run the following to get a nix-shell with the package available:
> 
> ```
> nix-shell -I nixpkgs=https://github.com/marsam/nixpkgs/archive/1966a0a4e1c1e07991406e3b90e3b088df252aea.tar.gz -p opencv4 python3Packages.opencv4
> ```

That is awesome! Thanks @mjlbach 

Using that I tested the change just now, and can confirm it works 👍  thanks @marsam for merging this in

**Pull request merged**